api/v1alpha1/releaseadmissionplan_webhook_test.go

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/release-service/kcp"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//+kubebuilder:scaffold:imports
+)
+
+var _ = Describe("ReleaseAdmissionPlan webhook", func() {
+	var releaseAdmissionPlan *ReleaseAdmissionPlan
+
+	BeforeEach(func() {
+		releaseAdmissionPlan = &ReleaseAdmissionPlan{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "ReleaseAdmissionPlan",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "releaseadmissionplan",
+				Namespace: "default",
+			},
+			Spec: ReleaseAdmissionPlanSpec{
+				DisplayName: "Test release plan",
+				Application: "application",
+				Origin: kcp.NamespaceReference{
+					Namespace: "default",
+				},
+				Environment:     "environment",
+				ReleaseStrategy: "strategy",
+			},
+		}
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, releaseAdmissionPlan)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	Context("When a ReleaseAdmissionPlan is created without the auto-release label", func() {
+		It("should get the label added with its value set to true", func() {
+			Expect(k8sClient.Create(ctx, releaseAdmissionPlan)).Should(Succeed())
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      releaseAdmissionPlan.Name,
+					Namespace: releaseAdmissionPlan.Namespace,
+				}, releaseAdmissionPlan)
+
+				labelValue, ok := releaseAdmissionPlan.GetLabels()[AutoReleaseLabel]
+
+				return err == nil && ok && labelValue == "true"
+			}, timeout).Should(BeTrue())
+		})
+	})
+
+	Context("When a ReleaseAdmissionPlan is created with an invalid auto-release label value", func() {
+		It("should get rejected until the value is valid", func() {
+			releaseAdmissionPlan.Labels = map[string]string{AutoReleaseLabel: "foo"}
+			err := k8sClient.Create(ctx, releaseAdmissionPlan)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("'%s' label can only be set to true or false", AutoReleaseLabel))
+		})
+	})
+
+	Context("When a ReleaseAdmissionPlan is created with a valid auto-release label value", func() {
+		It("shouldn't be modified", func() {
+			// Using value "true"
```

Comment:
I got to know today that ginkgo has the `By("Using value \"true\"")` definition to help to document and identify where a failure occurred in big `It` blocks. It might be useful in this case as you left this and the next comment.